src/java.base/share/classes/java/security/KeyPairGenerator.java

```diff
@@ -113,12 +113,16 @@
  * supply their own implementations of key pair generators.
  *
  * <p> Every implementation of the Java platform is required to support the
- * following standard {@code KeyPairGenerator} algorithms and keysizes in
- * parentheses:
+ * following standard {@code KeyPairGenerator} algorithms. For the "EC"
+ * algorithm, implementations must support the curves in parentheses. For other
+ * algorithms, implementations must support the key sizes in parentheses.
  * <ul>
- * <li>{@code DiffieHellman} (1024, 2048, 4096)</li>
+ * <li>{@code DiffieHellman} (1024, 2048, 3072, 4096)</li>
```

Comment:
Will there be a separate initiative to remove the 1024 from the requirements (or make clear that implementations are allowed to denn it by default policy).

I also wonder if more SHA3 should be added to the required algorithms.

The ticket Talks about tlsv1.3 and CNA 1.0, is the curve25519/chacha20 made mandatory due to TLS or just best practice?